docs/api-reference/next/image.md

```diff
@@ -497,6 +497,18 @@ The default [loader](#loader) will automatically bypass Image Optimization for a
 
 Auto-detection for animated files is best-effort and supports GIF, APNG, and WebP. If you want to explicitly bypass Image Optimization for a given animated image, use the [unoptimized](#unoptimized) prop.
 
+### Remote Cache Control
+
+By default, when an image came from a remote server the `Cache-Control` header is set as `public, max-age=0, must-revalidate`. To disable cache revalidation and change this behavior you can rewrite the header value by configuring `remoteCacheControl` option.
+
+```js
+module.exports = {
+  images: {
+    remoteCacheControl: 'public, max-age=315360000',
```

Comment:
Wondering something... Could it be a function too ? 

```typescript
remoteCacheControl: string | (originalUrl: string, urlReturnedByTheLoader: string) => string
```

To allow customization based on the given url (or the transformed one).

i.e:

```typescript
remoteCacheControl: (originalUrl: string, urlReturnedByTheLoader: string) => {
  if(isS3Url(url)) {
    return 'public, max-age=315360000';
  }
  // ...other cases
}
```

Just a concept, not a proper API... Might need more thoughts. Personally I would have preferred the loader function to return something like:

```typescript 
{ 
   url: 'transformed', 
   headers: Record<string, string>
}
```

But I guess it would be a bc.

Reply:
Would be cool allow it to be a function! We can have multiple sources, with different policies. The default could keep as the string (to avoid a breaking change).

Do you think this solution would work? I can quickly change it here :)

Reply:
@belgattitude I've checked the code and I'm rewriting my anwser:

I think we can let the `remoteCacheControl` variable be a string or a function, but not related with the loader, since the problem this PR tries to solve only happens when we use `next/image` to optimize/transform remote images. If we use a loader to point image to an external CDN without optimize it using next, the problem does not happens (since `next/image` points directly to external URL).

In any case, if we understand that `remoteCacheControl` is the way to go ahead with this pull request, I think that make sense let user decide the `Cache-Control` header based on remote image URL, that way we would be able to set different caching controls for each origin, like that:

```
remoteCacheControl: (url: string) => {
  if (isLocalImage(url)) {
    return 'public, max-age=0'
  }
  if(isS3Url(url)) {
    return 'public, max-age=315360000';
  } 
  // ...other cases
}
```

What do you guys think?

Reply:
I don't think a function is useful here because its not controlling Next.js built-in cache.

Furthermore, I don't think we should allow customizing the cache header for local images since those are hashed by content and already use the immutable header.

Reply:
@styfle Actually the example was not good, but we can have multiple remote locations, like a CDN from AWS, an image from API and any others, and the `cache-control` can be different for them, but the function works only for remote images.

I'll modify my code to allow another review.